front/src/api/role.api.ts

```diff
@@ -85,3 +85,35 @@ export const useUpdateRolePermissions = () => {
     },
   })
 }
+
+export const useDeleteRole = () => {
+  const queryClient = useQueryClient()
+
+  return useMutation({
+    ...window.tanstackApi.mutation(
+      'delete',
+      '/realms/{realm_name}/roles/{role_id}',
+      async (res) => res.json()
+    ).mutationOptions,
+    // FIXME: there is no bulk delete endpoint, and this one may be inefficient, and the
+    // stacked toast messages will look bad.
```

Comment:
Arguably there shouldn't be a bulk delete endpoint for this, but the issue in the comment is somewhat real. I'm no React magician by any means, but happy to hear suggestions.